src/neptune/internal/utils/git.py

```diff
@@ -90,3 +106,94 @@ def to_git_info(git_ref: Union[GitRef, GitRefDisabled]) -> Optional[GitInfo]:
         )
     except:  # noqa: E722
         return None
+
+
+class DiffTracker:
```

Comment:
This class does definitely too many things at once.  Try to prepare a dataclass that will handle the contents of diff and commit sha + a function that accepts a GitRef and returns this dataclass. `track_uncommitted_changes` shouldn't be aware of dirty vs non-dirty and `DiffTracker` currently handles data, computation and result at the same time.